protected/humhub/modules/admin/resources/js/humhub.admin.PendingRegistrations.js

```diff
@@ -0,0 +1,65 @@
+humhub.module('admin.PendingRegistrations', function(module, require, $) {
+    var object = require('util').object;
+    var Widget = require('ui.widget').Widget;
+    var client = require('client');
+    var urlDeleteSelected;
+
+    var PendingRegistrations = function(node, options) {
+        Widget.call(this, node, options);
+    };
+
+    object.inherits(PendingRegistrations, Widget);
+
+    PendingRegistrations.deleteAllSelected = function(evt) {
+        client.post(evt).then(function() {
+            var keys = $("#grid").yiiGridView("getSelectedRows");
+            $.ajax({
+                url: urlDeleteSelected,
+                type: "POST",
+                data: {id: keys},
+            })
+        }).catch(function(e) {
+            module.log.error(e, true);
+        })
+    };
+
+    PendingRegistrations.deleteAll = function(evt) {
```

Comment:
Change functions to:

`PendingRegistrations.prototype.deleteAll` and `PendingRegistrations.prototype.deleteAllSelected` otherwise those function won't be bound to your widget instance and are only available as static functions, also you won't be able to access `this`.

Then change the action buttons to:

`->action('deleteAll', Url::toRoute(['/admin/pending-registrations/delete-all']))`

and

  $('.delete-all').attr('data-action-click','deleteAllSelected');`

As long as those buttons reside in your widget, the widget function will be called.

Reply:
Done! Thank you!